src/referee.cpp

```diff
@@ -3099,6 +3110,14 @@ HFORef::analyse()
             {
                 M_take_time = 0;
             }
+
+            // Broadcast who has possession. Goalie catches are broadcast separately by the server. Rebroadcasting a message here would overwrite the other one. 
+            if (!(M_holder_side == 'R' && M_holder_unum == 1)){
```

Comment:
Let's break this into two lines: 
bool goalie_has_ball = M_holder_side == 'R' && M_holder_unum == 1;
if (!goalie_has_ball) { ...